kotlinx-coroutines-core/jvm/src/flow/TickerFlow.kt

```diff
@@ -0,0 +1,33 @@
+package kotlinx.coroutines.flow
+
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.awaitClose
+import java.util.*
+import kotlin.concurrent.schedule
+
+/**
+ * Creates a flow that produces the first item after the given initial delay and subsequent items with the
+ * given delay between them.
+ *
+ * The resulting flow is a callback flow, which basically listens @see [Timer.schedule]
+ *
+ * This Flow stops producing elements immediately after [Job.cancel] invocation.
+ *
+ * @param period period between each element in milliseconds.
+ * @param initialDelay delay after which the first element will be produced (it is equal to [period] by default) in milliseconds.
+ */
+public fun tickerFlow(
+        period: Long,
+        initialDelay: Long = period
```

Comment:
To follow the example of functions like `withTimeout` and `debounce`, I think we should add the `Millis` suffix to these parameter names, and maybe provide an overload using `kotlin.time.Duration`?

Reply:
Good point for naming and we have `.receiveAsFlow()` extension function.

Directly we can use kotlinx-coroutine's `ticker`. Also this can be better idea to have single implementation.
What do you think about it?

Reply:
It could indeed be done this way, but that would couple it to JVM platform 😥
I am not sure a channel would be necessary behind the scenes to implement it, but I will let the maintainers comment on this.

Also, I can see now why a concurrent behaviour is necessary and would like to backtrack a bit: we can't use a simple loop with delays because otherwise the ticker delay is impacted by the execution time of the collector code, which is probably not desirable in this case.

That's why we do need some concurrent process (a coroutine?) to send the ticks.